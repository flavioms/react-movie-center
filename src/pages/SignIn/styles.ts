import styled, { css } from 'styled-components';
import LogoPNG from '../../assets/logo.png';

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100vw;
  height: 100vh;
`;

export const Logo = styled.img.attrs(() => ({
  src: LogoPNG,
  alt: 'logo',
}))`
  width: 100%;
  max-width: 400px;
`;

export const SocialLogin = styled.div`
  display: flex;
  flex-direction: row;
  margin-top: 2em;

  @media only screen and (max-width: 460px) {
    flex-direction: column;
  }
`;

interface ButtonLoginProps {
  social?: string;
}

export const ButtonLogin = styled.button<ButtonLoginProps>`
  background: #fff;
  color: #fff;
  border: none;
  padding: 1em;
  display: flex;
  flex-wrap: nowrap;
  min-width: 200px;
  justify-content: flex-start;
  align-items: center;
  margin: 1em;
  ${props =>
    props.social === 'facebook' &&
    css`
      background: #3b5998;
    `}

  ${props =>
    props.social === 'google' &&
    css`
      background: #fff;
      color: #797979;
    `}

  & > svg {
    font-size: 1.6em;
    margin-right: 0.5em;
  }
`;
