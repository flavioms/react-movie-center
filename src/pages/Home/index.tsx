import React, { useContext } from 'react';
import AuthContext from '../../contexts/auth';

import { Container } from './styles';

const Home: React.FC = () => {
  const { signOutAuth } = useContext(AuthContext);
  return (
    <Container>
      <button type="button" onClick={() => signOutAuth()}>
        Sair
      </button>
    </Container>
  );
};

export default Home;
