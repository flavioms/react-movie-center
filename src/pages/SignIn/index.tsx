import React, { useContext, useEffect } from 'react';
import AuthContext from '../../contexts/auth';
import { Container } from './styles';

const SignIn: React.FC = () => {
  const { signed, signInAuth } = useContext(AuthContext);

  useEffect(() => {
    signInAuth();
  }, [signInAuth]);

  return (
    <Container>
      <form>
        {console.log(signed)}
        <input type="email" id="email" placeholder="Digite seu e-mail" />
        <input type="password" id="password" placeholder="Digite sua senha" />
        <button type="submit">Login</button>
      </form>
    </Container>
  );
};

export default SignIn;
