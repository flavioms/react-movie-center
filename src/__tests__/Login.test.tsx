import React, { useContext } from 'react';
import { shallow, mount } from 'enzyme';
import SignIn from '../pages/SignIn';
import Home from '../pages/Home';
import Routes from '../routes/index';
import AuthContext from '../contexts/auth';

describe('Testing SignIn Component', () => {
  it('render correctly', () => {
    const wrapper = shallow(<SignIn />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should have button login Facebook', () => {
    const wrapper = shallow(<SignIn />);
    const button = wrapper.find('#facebook');
    expect(button).toHaveLength(1);
  });

  it('should have button login Google', () => {
    const wrapper = shallow(<SignIn />);
    const button = wrapper.find('#google');
    expect(button).toHaveLength(1);
  });

  it('should show SingIn page', () => {
    const wrapper = mount(
      <AuthContext.Provider
        value={{
          signed: false,
          user: null,
          signInAuthFacebook: jest.fn(),
          signInAuthGoogle: jest.fn(),
          signOutAuth: jest.fn(),
        }}
      >
        <Routes />
      </AuthContext.Provider>,
    );

    expect(wrapper.find(SignIn)).toHaveLength(1);
  });

  it('should show Home page', () => {
    const wrapper = mount(
      <AuthContext.Provider
        value={{
          signed: true,
          user: { name: 'flavio', email: 'teste@teste.com.br' },
          signInAuthFacebook: jest.fn(),
          signInAuthGoogle: jest.fn(),
          signOutAuth: jest.fn(),
        }}
      >
        <Routes />
      </AuthContext.Provider>,
    );

    expect(wrapper.find(Home)).toHaveLength(1);
  });
});
