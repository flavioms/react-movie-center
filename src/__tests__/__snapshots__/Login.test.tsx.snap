// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SignIn Component render correctly 1`] = `
<styled.div>
  <button
    id="facebook"
    type="button"
  >
    Facebook
  </button>
  <button
    id="google"
    type="button"
  >
    Google
  </button>
</styled.div>
`;
